src/main/java/it/eng/idsa/dataapp/util/MessageUtil.java

```diff
@@ -77,4 +162,39 @@ private String createContractAgreement(Path dataLakeDirectory) {
 		}
 		return contractAgreement;
 	}
+	
+	private String getSelfDescription() {
+		URI eccURI = null;
+		try {
+			eccURI = new URI("http", null, eccProperties.getHost(), 
+					8081, null,
+					null, null);
+		} catch (URISyntaxException e) {
+			logger.error("Could not create URI for Self Description request.", e);
+		}
+		
+		return restTemplate.getForObject(eccURI, String.class);
+	}
+	
+	private String getRequestedElement(URI requestedElement) {
```

Comment:
Maybe to pass SelfDescription as parameter, instead making call internally.
And then change to containsRequestedElement or something like that

Reply:
Change the name?